src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/RuntimeBroker.cs

```diff
@@ -0,0 +1,399 @@
+            ICoreLogger logger,
+            IMsaPassthroughHandler msaPassthroughHandler = null)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _synchronizationContext = uiParent?.SynchronizationContext;
+
+            _parentHandle = GetParentWindow(uiParent);
+
+            _wamOptions = appConfig.WindowsBrokerOptions ??
+                WindowsBrokerOptions.CreateDefault();
+        }
+
+        /// <summary>
+        /// AcquireTokenInteractiveAsync
+        /// </summary>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <param name="acquireTokenInteractiveParameters"></param>
+        /// <returns></returns>
+        public async Task<MsalTokenResponse> AcquireTokenInteractiveAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+#if WINDOWS_APP
+            if (_synchronizationContext == null)
+            {
+                throw new MsalClientException(
+                    MsalError.WamUiThread,
+                    "AcquireTokenInteractive with broker must be called from the UI thread when using WAM." +
+                     ErrorMessageSuffix);
+            }
+#endif
+
+            _logger.Verbose("[WamBroker] Using Windows account picker.");
+
+            return await AcquireInteractiveWithPickerAsync(authenticationRequestParameters)
+                .ConfigureAwait(false);
+        }
+
+
+        /// <summary>
+        /// AcquireInteractiveWithPickerAsync
+        /// </summary>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <returns></returns>
+        /// <exception cref="MsalClientException"></exception>
+        /// <exception cref="MsalServiceException"></exception>
+        private async Task<MsalTokenResponse> AcquireInteractiveWithPickerAsync(AuthenticationRequestParameters authenticationRequestParameters)
+        {
+#if !UAP10_0
+            bool isMsaPassthrough = _wamOptions.MsaPassthrough;
+            var authority = authenticationRequestParameters.Authority.AuthorityInfo.CanonicalAuthority;
+            MsalTokenResponse tokenResponse = null;
+
+            NativeInterop.AuthResult result = null;
+
+            try
+            {
+                if (WindowsNativeUtils.IsElevatedUser())
+                {
+                    _logger.Verbose($"[WamBroker] Current user is the member of the admin group. WAM should still work.");
+                }
+
+                using (var core = new NativeInterop.Core())
+                using (var authParams = new NativeInterop.AuthParameters(authenticationRequestParameters.AppConfig.ClientId, authority))
+                {
+                    authParams.RequestedScopes = string.Join(" ", authenticationRequestParameters.Scope);
+                    authParams.RedirectUri = authenticationRequestParameters.RedirectUri.ToString();
```

Comment:
I wonder why this is needed, since WAM redirect uri does not need to be added to the app. Maybe for MSAL C++'s own fallback to browser?